lib/internal/Magento/Framework/Webapi/ServiceInputProcessor.php

```diff
@@ -519,6 +520,13 @@ public function convertValue($data, $type)
                     foreach ($data as $key => $item) {
                         $result[$key] = $this->_createFromArray($itemType, $item);
                     }
+                } else {
```

Comment:
Please add some automated tests with a negative scenario where we are not passing the array.